python/tests/test_tables.py

```diff
@@ -4510,19 +4510,15 @@ def verify_subset(self, tables, nodes):
         for k, p in zip(pops, subset.populations):
             pp = tables.populations[k]
             assert pp == p
-        edges = [
-            i
-            for i, e in enumerate(tables.edges)
+        # subset can reorder the edges: we need to check we have the same set
+        edges = {
+            e.replace(parent=node_map[e.parent], child=node_map[e.child])
+            for e in tables.edges
             if e.parent in nodes and e.child in nodes
-        ]
+        }
         assert subset.edges.num_rows == len(edges)
-        for k, e in zip(edges, subset.edges):
-            ee = tables.edges[k]
-            assert ee.left == e.left
-            assert ee.right == e.right
-            assert node_map[ee.parent] == e.parent
-            assert node_map[ee.child] == e.child
-            assert ee.metadata == e.metadata
+        for e in subset.edges:
+            assert e in edges
```

Comment:
This is a subtly looser check as now there can be edges in `edges` that aren't in `subset.edges`. I guess actually that could happen before as the zip would stop at the shortest. Adding a paranoid:
```
for e in edges:
    assert e in subset.edges
```
Would make it strict?